earth_enterprise/src/fusion/portableglobe/rewritedbroot_unittest.cpp

```diff
@@ -0,0 +1,166 @@
+// Copyright 2017 the Open GEE Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include "rewritedbroot.h"
+#include <string>
+
+using namespace std;
+
+int translationEntryId = 0;
+
+TEST(ReplaceReferencedKmlTest, NoKmlTest) {
+  geProtoDbroot dbroot;
+  string before = dbroot.ToTextString();
+  ReplaceReferencedKml("http://localhost", "", false, &dbroot);
+  string after = dbroot.ToTextString();
+  ASSERT_EQ(before, after);
+}
+
+void addKmlUrlAsValue(const string & url, geProtoDbroot * dbroot) {
+  keyhole::dbroot::NestedFeatureProto * nested = dbroot->add_nested_feature();
+  keyhole::dbroot::StringIdOrValueProto * kmlUrl = nested->mutable_kml_url();
+  kmlUrl->set_value(url);
+}
+
+TEST(ReplaceReferencedKmlTest, NestedEntryReplaceTest) {
+  const string BEFORE_URL_1 = "http://server.name/kmlfile.kmz";
+  const string BEFORE_URL_2 = "http://other.server.name/newfile.kml";
+  const string AFTER_BASE_URL = "https://different.server";
+  const string AFTER_URL_1 = AFTER_BASE_URL + "/dbroot_kml_000.kml";
+  const string AFTER_URL_2 = AFTER_BASE_URL + "/dbroot_kml_001.kml";
+  geProtoDbroot dbroot;
+  
+  addKmlUrlAsValue(BEFORE_URL_1, &dbroot);
+  dbroot.add_nested_feature();
+  addKmlUrlAsValue(BEFORE_URL_2, &dbroot);
+  dbroot.add_nested_feature();
+  dbroot.add_nested_feature();
+
+  ASSERT_TRUE(dbroot.nested_feature(0).has_kml_url());
+  ASSERT_EQ(dbroot.nested_feature(0).kml_url().value(), BEFORE_URL_1);
```

Comment:
Suggestion: Check the value of `BEFORE_URL_2`, as well.

Reply:
I didn't do that because this is not really what we're testing - I'm just verifying that I set up the test correctly. However, I can see the point of checking everything here, too.